include/exadg/structure/preconditioners/multigrid_preconditioner.h

```diff
@@ -59,14 +59,16 @@ class MultigridPreconditioner : public MultigridPreconditionerBase<dim, Number>
   MultigridPreconditioner(MPI_Comm const & mpi_comm);
 
   void
-  initialize(MultigridData const &                       mg_data,
-             dealii::Triangulation<dim> const *          tria,
-             dealii::FiniteElement<dim> const &          fe,
-             std::shared_ptr<dealii::Mapping<dim> const> mapping,
-             ElasticityOperatorBase<dim, Number> const & pde_operator,
-             bool const                                  nonlinear_operator,
-             Map const &                                 dirichlet_bc,
-             PeriodicFacePairs const &                   periodic_face_pairs);
+  initialize(
+    MultigridData const &                                                  mg_data,
+    dealii::Triangulation<dim> const *                                     tria,
+    std::vector<std::shared_ptr<dealii::Triangulation<dim> const>> const & coarse_triangulations,
```

Comment:
Maybe we could pass in `Grid` directly here?

Reply:
I actually followed the same goal and I tried this today in the morning, but gave it up in the end because there are places in the code where the mapping parameter passed to multigrid is `get_dynamic_mapping(mapping, moving_mesh)`. Hence, this problem is better addressed separately.